src/main/java/org/terasology/projsndwv/TempBeeRegistry.java

```diff
@@ -15,26 +15,114 @@
  */
 public class TempBeeRegistry {
-    @In
-    public static AssetManager assetManager;
+    private static HashMap<Integer, Integer> lifespans = new HashMap<>();
+    private static HashMap<Integer, Long> tickspeeds = new HashMap<>();
+    private static HashMap<Integer, Production> productions = new HashMap<>();
+
+    private static MersenneRandom random;
+
+    private static Logger logger = LoggerFactory.getLogger(TempBeeRegistry.class);
 
     public static TextureRegionAsset<?> getTextureRegionAssetForSpeciesAndType(int species, int type) {
-        return (TextureRegionAsset<?>)CoreRegistry.get(AssetManager.class).getAsset("Apiculture:bee_" + new String[] {"a", "b", "c", "d"}[species - 1] + "_" + new String[] {"drone", "princess", "queen"}[type], TextureRegionAsset.class).get();
+        AssetManager assetManager = CoreRegistry.get(AssetManager.class);
+        if (assetManager == null) {
+            logger.error("No AssetManager in registry");
+            return null;
+        }
+
+        // TODO: This call can probably be checked, but it'll take a serious refactor.
+        Optional<TextureRegionAsset> textureRegionAsset = assetManager.getAsset("Apiculture:bee_" + new String[]{"a", "b", "c"}[species] + "_" + new String[]{"drone", "princess", "queen"}[type], TextureRegionAsset.class);
+        if (!textureRegionAsset.isPresent()) {
+            logger.error("Texture not found for species " + species + " and type " + type);
+            return null;
+        }
+
+        return textureRegionAsset.get();
     }
 
     public static DisplayNameComponent getDisplayNameComponentForSpeciesAndType(int species, int type) {
         DisplayNameComponent displayNameComponent = new DisplayNameComponent();
-        displayNameComponent.name = new String[] {"A", "B", "C", "D"}[species - 1] + " " + new String[] {"Drone", "Princess", "Queen"}[type];
+        displayNameComponent.name = new String[] {"A", "B", "C"}[species] + " " + new String[] {"Drone", "Princess", "Queen"}[type];
         return displayNameComponent;
     }
+
+    public static int getLifespanFromGenome(int genome) {
+        return lifespans.get(genome);
+    }
+
+    public static long getTickTimeFromGenome(int genome) {
+        return tickspeeds.get(genome);
+    }
+
+    public static EntityRef getProduceForSpeciesWithChance(int species) {
+        MersenneRandom random = getRandom();
+        Production production = productions.get(species);
+
+        if (Objects.requireNonNull(random).nextFloat() < production.chance) {
+            EntityManager entityManager = CoreRegistry.get(EntityManager.class);
+            if (entityManager == null) {
+                logger.error("No EntityManager in registry");
+                return null;
+            }
+            return entityManager.create(production.prefab);
+        }
+        else {
+            return EntityRef.NULL;
+        }
+    }
+
+    private static MersenneRandom getRandom() {
```

Comment:
Is there a particular reason you are trying to tie the random here to the world seed?
Only the world gen itself needs to be deterministic, the gameplay does not.

Reply:
Arbitrary design choice. It's a traditional feature.